### ISSUE TITLE
--fast parameter makes Gifski crash under windows

### ISSUE THREAD

**Issue by jalba69** [ImageOptim/gifski#24]
Hello,

**--fast** parameter seems to crash Gifski 0.7.3 ( release from releases page on git ) under windows, ( tested with win7 x64 )

Checked the options with -h, the --fast switch is indicated : 
![2018-01-24_132405_cr](https://user-images.githubusercontent.com/15131985/35332329-ec047144-010a-11e8-9ef4-4572103d36ff.png)
( btw: seems like there is a little typo for the _--quiet_ description, _"Do**n** not"_ instead of _"Do not"_ i suppose )

With command : 
`gifski -o anim.gif --fast "H:\Gif_Frames\frame"*".png"`
or when frames are in the same folder : 
`gifski -o anim.gif --fast frame*.png`
or with FPS added like here 
`gifski -o anim.gif --fps 30 --fast "H:\Gif_Frames\frame"*".png"`

The first frame of the gif is created then I get a crash and a memory related error : 
![2018-01-24_133044](https://user-images.githubusercontent.com/15131985/35332332-f043cb60-010a-11e8-9c63-22d80db141f2.png)
Translation: _Instruction @ 0x3fa5a262 use memory address 0x00000000. The memory can't be in "written" state._

No problems at all to create gif's when not using --fast option and for example a command like : 
`gifski -o anim.gif --fps 30 --quality 100 "H:\Gif_Frames\frame"*".png"`
This works fine.

Also i don't know where to ask so forgive me if it's not the right place or send me to the appropriate place to ask about this : 

I use the official released binaries, i can't compil stuff myself and the few times i tried to compile things it ended in headaches and failures... but i'm interested in the version able to use directly videos for input via ffmpeg and i'm wondering about a few things : 

- Does it supports all input formats ffmpeg support ( including avisynth .avs extension files ? )
- Is there a difference in the gif output size, i mean if we take a video and input it directly vs a video that we first convert to png's is there any differences at all ?
- related to above : is ffmpeg "piping" directly the video to gifski or decomposing the input video into png's frames internally before feeding gifski. 

Not sure if you or someone else have tested to see if there is a difference of size ( and quality maybe ? ) of the output gif using the 2 methods.

Like i said i can't really compile stuff, but before spending again half a week trying to get gifski compiled to work with videos ( and probably failing miserably at it in the end ) i would like to know if when using ffmpeg there is a real difference and how frames are sent to gifski, would particularly be interested in feeding it directly with .avs files since ffmpeg supports it.

Thank you very much.

**Comment by kornelski** [comment]
Differences between video and png frames:

- the decoder may use different way of smoothing chroma channel, causing saturated red/pink/blue to be either more blocky or more blurry. It's best if you make GIF half resolution of video input.

- direct video decoding uses "presentation timestamp" of each frame to control framerate, which in theory is more accurate, but in practice there are lots of messed up videos with jumpy framerate.

- direct video encoding avoids and extra step of temporary png files, so it's a bit faster. 

Other than that there's no difference.

**Comment by jalba69** [comment]
Thank you for the answers,

beside skipping the creation of png's wich is a very good thing,  it's interesting, seems like all the rest can be controlled easily on the video side, since it's stuff i encode myself there is no jumpy FPS or other weird things from random videos.

Will be even more easy if gifsky compiled with ffmpeg accepts .avs scripts for input like ffmpeg does natively, this will make it a dream to work with, triming ,resizing, change fps of the videos or apply some filters, or add subtitles/captions, or overlays directly and quickly in the avisynth .avs script without the need to encode the video itself before feeding gifski, just feeding it with the .avs script file and ffmpeg ( should ) take care of it like it does usually with avs files.

I guess i will need to spend way more time trying to get something compiled successfully when i'll have a few days to burn on this, for now compiling things is the bane of my existence so i won't be able to test this anytime soon i think, if ever... :/  

As for the --fast parameter making gifski crash, any idea on what is making it crash on windows please ?

Thanks a lot for having took the time to answer.

**Comment by kornelski** [comment]
Regarding the writing problem, I think I've found the cause - before unzipping, click properties of the ZIP file and check "Unblock" checkbox.

**Comment by kornelski** [comment]
I know nothing about avs scripts. If they don't work already, then they won't. ffmpeg is PITA to build and redistribute. I'm considering using gstreamer instead.

**Comment by jalba69** [comment]
Hello, 

Sorry but i have no "Unblock" checkbox when looking at proprieties under windows seven. I use winrar to extract the archive, no issues on this side either.

The exe have rights to read, write execution and all needed, running it as admin but it + the console command with admin rights doesn't change anything for me. It crashes after creating the first frame o the gif when using --fast parameter.

Like i said no problems when not using this --fast parameter.

.avs can be read natively by ffmpeg, i guess it should work but not having a ffmpeg compiled version of gifski i cannot test sadly and seeing your comment i guess it will be another failure for me on my list of compiling attempts if i try to do it myself... basically and .avs is just a script file for avisynth that ffmpeg can deal with but without a gifski.exe compiled with ffmpeg hard to tell if it is supported and working as supposed :p 

Thank you.

**Comment by kornelski** [comment]
Unfortunately I'm unable to reproduce the crash on my machine. The 0.7.3 gifski.exe works with the --fast flag just as well as without. The error message is very generic and doesn't tell me what's going on.

**Comment by kornelski** [comment]
Building gifski without ffmpeg should be relatively simple. Once you have Rust and Visual Studio installed, `cargo build` should "just work". Try that and see if executable built on your machine crashes too.

**Comment by jalba69** [comment]
Hello, 

spent some time on this, installed visual studio community 2017 and Rust, managed to get basic Gifski.exe built, 
![2018-01-25_161555](https://user-images.githubusercontent.com/15131985/35396330-d646a9fc-01ec-11e8-8c47-f1c73f768f73.png)
finally achieved compiling something myself, hoorah :p 

Gave some tries, when i start to use --fast it crashes, without this option it works fine, like the official releases.

When it crashed it told me it could now open the debug in visual studio, I'm not too familiar with this but this is what i saw there : 
![2018-01-25_155937](https://user-images.githubusercontent.com/15131985/35396374-f55cd50a-01ec-11e8-9d78-d39ed7d93bc5.png)

Same stuff in plain text : 
_" Exception non gérée à 0x000000013F34A2E2 dans gifski.exe : 0xC0000005 : Violation d'accès lors de l'écriture à l'emplacement 0x0000000000000000. a eu lieu "_
```
'gifski.exe' (Win32) : Chargé 'H:\gifski.exe'. Les symboles ont été chargés.
'gifski.exe' (Win32) : Chargé 'C:\Windows\System32\ntdll.dll'. Impossible de trouver ou d'ouvrir le fichier PDB.
'gifski.exe' (Win32) : Chargé 'C:\Windows\System32\kernel32.dll'. Impossible de trouver ou d'ouvrir le fichier PDB.
'gifski.exe' (Win32) : Chargé 'C:\Windows\System32\KernelBase.dll'. Impossible de trouver ou d'ouvrir le fichier PDB.
'gifski.exe' (Win32) : Chargé 'C:\Windows\System32\advapi32.dll'. Impossible de trouver ou d'ouvrir le fichier PDB.
'gifski.exe' (Win32) : Chargé 'C:\Windows\System32\msvcrt.dll'. Impossible de trouver ou d'ouvrir le fichier PDB.
'gifski.exe' (Win32) : Chargé 'C:\Windows\System32\sechost.dll'. Impossible de trouver ou d'ouvrir le fichier PDB.
'gifski.exe' (Win32) : Chargé 'C:\Windows\System32\rpcrt4.dll'. Impossible de trouver ou d'ouvrir le fichier PDB.
'gifski.exe' (Win32) : Chargé 'C:\Windows\System32\dbghelp.dll'. Impossible de trouver ou d'ouvrir le fichier PDB.
'gifski.exe' (Win32) : Chargé 'C:\Windows\System32\vcruntime140.dll'. Impossible de trouver ou d'ouvrir le fichier PDB.
'gifski.exe' (Win32) : Chargé 'C:\Windows\System32\api-ms-win-crt-runtime-l1-1-0.dll'. Impossible de trouver ou d'ouvrir le fichier PDB.
'gifski.exe' (Win32) : Chargé 'C:\Windows\System32\ucrtbase.dll'. Impossible de trouver ou d'ouvrir le fichier PDB.
'gifski.exe' (Win32) : Chargé 'C:\Windows\System32\api-ms-win-core-timezone-l1-1-0.dll'. Impossible de trouver ou d'ouvrir le fichier PDB.
'gifski.exe' (Win32) : Chargé 'C:\Windows\System32\api-ms-win-core-file-l2-1-0.dll'. Impossible de trouver ou d'ouvrir le fichier PDB.
'gifski.exe' (Win32) : Chargé 'C:\Windows\System32\api-ms-win-core-localization-l1-2-0.dll'. Impossible de trouver ou d'ouvrir le fichier PDB.
'gifski.exe' (Win32) : Chargé 'C:\Windows\System32\api-ms-win-core-synch-l1-2-0.dll'. Impossible de trouver ou d'ouvrir le fichier PDB.
'gifski.exe' (Win32) : Chargé 'C:\Windows\System32\api-ms-win-core-processthreads-l1-1-1.dll'. Impossible de trouver ou d'ouvrir le fichier PDB.
'gifski.exe' (Win32) : Chargé 'C:\Windows\System32\api-ms-win-core-file-l1-2-0.dll'. Impossible de trouver ou d'ouvrir le fichier PDB.
'gifski.exe' (Win32) : Chargé 'C:\Windows\System32\api-ms-win-crt-string-l1-1-0.dll'. Impossible de trouver ou d'ouvrir le fichier PDB.
'gifski.exe' (Win32) : Chargé 'C:\Windows\System32\api-ms-win-crt-heap-l1-1-0.dll'. Impossible de trouver ou d'ouvrir le fichier PDB.
'gifski.exe' (Win32) : Chargé 'C:\Windows\System32\api-ms-win-crt-stdio-l1-1-0.dll'. Impossible de trouver ou d'ouvrir le fichier PDB.
'gifski.exe' (Win32) : Chargé 'C:\Windows\System32\api-ms-win-crt-convert-l1-1-0.dll'. Impossible de trouver ou d'ouvrir le fichier PDB.
'gifski.exe' (Win32) : Chargé 'C:\Windows\System32\api-ms-win-crt-math-l1-1-0.dll'. Impossible de trouver ou d'ouvrir le fichier PDB.
'gifski.exe' (Win32) : Chargé 'C:\Windows\System32\api-ms-win-crt-utility-l1-1-0.dll'. Impossible de trouver ou d'ouvrir le fichier PDB.
'gifski.exe' (Win32) : Chargé 'C:\Windows\System32\api-ms-win-crt-locale-l1-1-0.dll'. Impossible de trouver ou d'ouvrir le fichier PDB.
'gifski.exe' (Win32) : Chargé 'C:\Windows\System32\cryptbase.dll'. Impossible de trouver ou d'ouvrir le fichier PDB.
'gifski.exe' (Win32) : Chargé 'C:\Windows\System32\apphelp.dll'. Impossible de trouver ou d'ouvrir le fichier PDB.
Le thread 0x187c s'est arrêté avec le code 0 (0x0).
Exception non gérée à 0x000000013F489612 dans gifski.exe : 0xC0000005 : Violation d'accès lors de l'écriture à l'emplacement 0x0000000000000000.

```
_" Impossible de trouver ou d'ouvrir le fichier PDB. "_ translated means : _Impossible to find or open the file PDB_

I checked for some of the files paths listed and they are present in the path indicated and should be accessible fro ma permissions point of view, 
but they are also present in the " sysWOW64 " folder that we find in windows x64 builds too, and here i see this _gifski.exe' (**Win32**)_ not sure if it should be this or win64 and check into sysWOW64 folder instead or if it's normal to see win32, i have no clue, this bypass largely my little knowledge...

Don't know if this can help a bit or not to identify a potential issue but that's where i get for now.

I don't know what _--fast_ is supposed to do on the process when it is turned ON, might be Windows DEP preventing some memory execution or something else, i don't know, and without having an idea about what --fast parameter changes in the way gifski works it's hard to guess and dig more into this.

I'm on win7 x64 Professional SP1 in French language,  if some others having win7 x64 could test this one of those day when passing around and seeing this report to see if it's on my side only or something specific to this OS... Never encountered a problem like this with all the tools i run on this PC since many years, first time i see such issue, and here it seems that it's only this --fast parameter that doesn't work, the rest seems to work fine which makes it even more weird.


Next step is trying to get ffmpeg files needed and compile with video feature, haven't looked much into this for now to know what is exactly needed or where to put the ffmpeg files etc but i will focus on that tomorrow and probably this week-end as i will probably not succeed on the first tries, now that i reinstalled all the things needed maybe with a bit o luck i will manage to get this done.

Thank you very much for the help and patience.

**Comment by kornelski** [comment]
Build it again without the `--release` flag. It will use the debug mode, which prints more useful info.

**Comment by jalba69** [comment]
In console this is what i get when i try : 
H:\>gifski -o anim.gif --fast "H:\Gif_Frames\frame"*".png"
```
Frame 1 / 256  _............................................................................. 2m 
Assertion failed: img->temp_f_row, file msvc-dist/libimagequant.c, line 831
```

**Comment by kornelski** [comment]
Oh, great! How that's something I can fix :)

**Comment by jalba69** [comment]
Glad to see this, was starting to be worried about something weird happening only my end and my machine with no explanations of why it happens.

Thanks a lot.

**Comment by jalba69** [comment]
I'm not sure if your answer is ironic or not :p

If it isn't then i'm glad to see this , was starting to be worried about something weird happening only my end and my machine with no explanations of why it happens.

If it is well nothing much to do then i guess.
 
Thanks a lot.

**Comment by kornelski** [comment]
No, honestly that's a useful hint.

**Comment by jalba69** [comment]
My English sometimes fails me so i wasn't sure. :p 
That's a good news then :)
no need to hurry for a potential fix too, take your time.

**Comment by jalba69** [comment]
After spending the last 10 days or so trying various things i got the fast parameter to work... partially only.

I did nothing on my end that could really change and make it work, i believe that some packages downloaded by rust have changed a bit maybe; i don't have a real clues.

Anyways this seems to works now either with regular --release i compiled or compiling using --features=openMP, all those are OK :  
`## Normal ###
gifski -o anim.gif --fast "H:\Frames\frame"*".png"

gifski -o anim.gif --fps 24 --quality 100 "H:\Frames\frame"*".png"

gifski -o anim.gif --fps 24 --quality 100 --width 580 "H:\Frames\frame"*".png"

gifski -o anim.gif --fps 60 --quality 100 --width 600 --height 400 "H:\Frames\frame"*".png"`

`###` Fast ### 
gifski -o anim.gif --fast "H:\Frames\frame"*".png"

gifski -o anim.gif --fps 24 --fast "H:\Frames\frame"*".png"

gifski -o anim.gif --fast --fps 24 "H:\Frames\frame"*".png"

But if i throw in the resize parameter it crash again like before, all those commands lead to a crash : 
`### Crash ###
gifski -o anim.gif --width 480 --fast "H:\Frames\frame"*".png"           <--- Crash 

gifski -o anim.gif --fast --width 480 "H:\Frames\frame"*".png"           <--- Crash 

gifski -o anim.gif --fps 24 --fast --width 480 "H:\Frames\frame"*".png"  <--- Crash 

gifski -o anim.gif --fps 24 --width 480 --fast "H:\Frames\frame"*".png"  <--- Crash 

gifski -o anim.gif --fast --fps 24 --width 480 "H:\Frames\frame"*".png"  <--- Crash

gifski -o anim.gif --fast --width 480 --fps 24 "H:\Frames\frame"*".png"  <--- Crash

gifski -o anim.gif --width 480 --fps 24 --fast "H:\Frames\frame"*".png"  <--- Crash

gifski -o anim.gif --width 480 --fast --fps 24 "H:\Frames\frame"*".png"  <--- Crash`

As you can see i tried all combinations possible for the switches, just in case...

Error message from the debug version is the same as before : 
`H:\>gifski -o anim.gif --fast --width 600 "H:\Frames\frame"*".png"
Frame 1 / 256  _............................................................................ 18m Assertion failed: img->temp_f_row, file msvc-dist/libimagequant.c, line 831`

It won't bother me much as i will do the resizing before making frames so i think i will simply stop to bother with this for now.

- As for including the possibility to include video as input using FFmpeg and --releases=video... that's where i spent most of my free time for the last ten days or so, doing a lot of searches and reading loads of things, many of the things i found on the net had you involved in the discussion btw, i see you passed by this too but well i'm far from being a coder so it was harder for me :p 

 Building Rust-FFMpeg that is needed just seems not possible in windows using VS, once i get to this step it becomes a blocker... 
Then tried to build Rust-FFMpeg in standalone, without Gifski involved, using the files from them git and always failed, tried so many things while trying to make it work that i lost count of attempts and hours lost in this... 
Ended up seeing one discussion, where you and the Dev from Gstreamer (i think it was him) was answering someone having similar issue, seems like Rust-FFmpeg is just not suited for windows...

I will probably have to ask on the Rust-FFmpeg and see what can be done so this thing can be built under windows, once this possible, if ever, then i might be able to build Gifski with ffmpeg input videos, which seems like something fantastic for this tool as it seems to support .avs avisynth scripts as direct input bringing loads of possibilities without having to touch/re-code the videos themselves.

Like i said i spent lot of time trying things, way too much, in my attempts i found out vcpkg tool to manage the libraries on visual studio and this alone is making it not a total loss of time, probably Rust-FFmpeg could be built using directly the libraries managed by vcpkg on windows and this will make it way easier to manage on windows systems... i guess.

All this ended like i expected, compiling on windows is a headache, yet this time i learned some new things and probably won't uninstall all the tools i had to install to test all this and will start giving a few tries to some other tools i use... ( heck even tried something with mingw-64 in one of my attempts... had to use a freaking VPN just to to install it as this things years later still have not fixed the silly problem of Sourceforge using loads of different mirrors for downloads and making it impossible to install at the start because it needs to download something online during the start of the install process and cannot find it, had to use a German IP to get this thing installed after spending hours trying to find why it did not worked, many have the same issue... ) 

Well that's it for me, i really regret to not have Gifski supporting the videos and .avs scripts input via FFmpeg but at this point there isn't much i can do, I'll ask on the Rust-FFmpeg project to see but after spending some much time on this i feel burned up now, using vcpkg to manage the libraries seems like a very easy way to get this thing handled for windows users but I don't really have hope to see things improved for windows and this Rust package, I'll ask there, who knows...

Thanks a lot for all, gifski is usable and i will use it often even if not at his full potential due to the last crashes situations i reported or worse the lack of input videos and potentially the use of avisynth scripts directly which will just be a dream, if i understood correctly due to licenses reasons you won't release binaries supporting this, so unless i find one already compiled with --features=video somewhere or manage to get all Rust packages to build correctly under windows it's a lost cause :/ 

Thank you again for your nice tools,
( And sorry for this long post :p )

**Comment by jalba69** [comment]
One very last thing, i doubt it can help but who knows, when now crashing and using VS2017 debugger i see this 

`000000013FB3A292  movups      xmmword ptr [rsi+r8*8],xmm3 `
and 
`gifski.exe!__local_stdio_printf_options()`

![2018-02-04_145729](https://user-images.githubusercontent.com/15131985/35778319-f0a6a20c-09bc-11e8-8bc8-22c47a44d870.png)

![2018-02-04_145736](https://user-images.githubusercontent.com/15131985/35778320-f9417248-09bc-11e8-87bf-37cfe3ed0b2f.png)

and this : 
![2018-02-04_144843](https://user-images.githubusercontent.com/15131985/35778322-fddcf124-09bc-11e8-80f9-b7c11046bfa4.png)
This last one says something like " Octal Number not recognized" 

For me trying to understand this is like trying to read hieroglyphs but just in case i leave it here, who knows...

Thanks.

**Comment by kornelski** [comment]
Thanks for your patience on this. I think I've fixed the bug in imagequant library that was causing the crash. Please try:

```rust
cargo update
cargo build
```

and see if that works.

**Comment by jalba69** [comment]
Hello,
no need to thank me, it's me who have to thank you for all those nice tools ;)

Updated cargo :
![2018-02-05_174456](https://user-images.githubusercontent.com/15131985/35818921-9332f658-0aa1-11e8-9349-0c115302789a.png)

Built debug version and run those :
```
### Normal ###
gifski -o anim.gif --fps 24 --quality 100 "H:\Frames\frame"*".png"
### Fast ###
gifski -o anim.gif --fast "H:\Frames\frame"*".png"
gifski -o anim.gif --fast --fps 24 "H:\Frames\frame"*".png"
gifski -o anim.gif --width 480 --fast "H:\Frames\frame"*".png"
gifski -o anim.gif --width 480 --fast --fps 24 "H:\Frames\frame"*".png"
```
The ones that used to crash works without crash now.

Then built --release version and tried again quickly a couple o commands, no crashes :)

Then tried to build with openmp and...  no dice,  gives me this backtrace :
`H:\gifski-0.8.0>cargo clean

H:\gifski-0.8.0>cargo build --release --features=openmp
   Compiling ansi_term v0.10.2
   Compiling rgb v0.8.7
   Compiling cfg-if v0.1.2
   Compiling winapi v0.2.8
   Compiling openmp-sys v0.1.2
   Compiling imagequant-sys v2.11.9
   Compiling rayon v0.9.0
error: failed to run custom build command for `imagequant-sys v2.11.9`
process didn't exit successfully: `H:\gifski-0.8.0\target\release\build\imagequant-sys-7945b427deb23
0e6\build-script-build` (exit code: 101)
--- stderr
thread 'main' panicked at '

Internal error occurred: Command "C:\\Program Files (x86)\\Microsoft Visual Studio\\2017\\Community\
\VC\\Tools\\MSVC\\14.12.25827\\bin\\HostX64\\x64\\cl.exe" "/nologo" "/MD" "/O2" "/openmp" "-msse" "/
DNDEBUG=1" "/DUSE_SSE=1" "/FoH:\\gifski-0.8.0\\target\\release\\build\\imagequant-sys-642820e84d7208
32\\out\\msvc-dist\\mediancut.o" "/c" "msvc-dist/mediancut.c" with args "cl.exe" did not execute suc
cessfully (status code exit code: 2).

', C:\Users\USER\.cargo\registry\src\github.com-1ecc6299db9ec823\cc-1.0.4\src\lib.rs:1984:4
stack backtrace:
   0: std::sys_common::backtrace::_print
             at C:\projects\rust\src\libstd\sys_common\backtrace.rs:91
   1: std::panicking::default_hook::{{closure}}
             at C:\projects\rust\src\libstd\panicking.rs:380
   2: std::panicking::default_hook
             at C:\projects\rust\src\libstd\panicking.rs:397
   3: std::panicking::rust_panic_with_hook
             at C:\projects\rust\src\libstd\panicking.rs:577
   4: std::panicking::begin_panic<alloc::string::String>
             at C:\projects\rust\src\libstd\panicking.rs:538
   5: std::panicking::begin_panic_fmt
             at C:\projects\rust\src\libstd\panicking.rs:522
   6: cc::Tool::to_command
   7: cc::Build::compile
   8: ONCE
   9: panic_unwind::__rust_maybe_catch_panic
             at C:\projects\rust\src\libpanic_unwind\lib.rs:101
  10: std::rt::lang_start
             at C:\projects\rust\src\libstd\rt.rs:51
  11: __scrt_common_main_seh
             at f:\dd\vctools\crt\vcstartup\src\startup\exe_common.inl:283
  12: BaseThreadInitThunk

warning: build failed, waiting for other jobs to finish...
`
Not sure what it is about ...

Either _cargo build --release --features=openmp_ or just _cargo build features=openmp_ crash on this last released version.
Was building openmp without errors on previous version, not sure if it was then working correctly or not tho, never took the time to time out the gif construction.

Will run a few more tests later at night ( and probably tomorrow), see the results then report back on this, (and also give another try for the ffmpeg without any hope for this one :p )

For now beside openmp failing to build ( might be normal ) it seems that it works ^^, no more crashes so it's very nice, thanks a lot for fixing it :) 

Thanks a lot for your time and patience, i'll do more testings and report back later at night or most probably tomorrow after doing a few more gifs.

**Comment by kornelski** [comment]
Try `cargo build -vvv --features=openmp` to get more information. Previously the `--features=openmp` argument did nothing on Windows :)

**Comment by jalba69** [comment]
Good to kow it wasn't doing anything, on the main page it was indicated for linux and MACos , i tried but never timed the results to confirm ^^ VS studio have some OpenMp settings so i tried...

Done the command with backtrace and log to file like : 
`cargo build -vvv --features=openmp >> log.txt`

log is a bit long so i put here here : 
https://pastebin.com/raw/0R0X2yQA

I'm on a French setup, looks like UFT8 characters are not passed correctly to the log so you will see some weird � symbols:

`cargo:warning=cl�: Ligne de commande warning D9002�: option '-msse' inconnue ignor�e`
Means option '-msse' unknown, will be ignored.

`msvc-dist/mediancut.c(198): error C3015: forme incorrecte de l'initialisation dans l'instruction 'for' OpenMP
msvc-dist/mediancut.c(213): error C3015: forme incorrecte de l'initialisation dans l'instruction 'for' OpenMP
msvc-dist/mediancut.c(314): error C3001: 'task'�: nom de directive OpenMP attendu
msvc-dist/mediancut.c(316): error C3001: 'task'�: nom de directive OpenMP attendu
msvc-dist/mediancut.c(341): error C3001: 'taskgroup'�: nom de directive OpenMP attendu
msvc-dist/mediancut.c(388): error C3001: 'taskgroup'�: nom de directive OpenMP attendu
msvc-dist/mediancut.c(445): error C3015: forme incorrecte de l'initialisation dans l'instruction 'for' OpenMP
exit code: 2`

c3015 refers to incorrect initialization of the instruction 'for' OpenMp
copy pasted it in google and i sa some similar results :
https://docs.microsoft.com/en-us/cpp/error-messages/compiler-errors-2/compiler-error-c3015

c3001 refers to name of guidance OpenMP awaited ,
https://docs.microsoft.com/en-us/cpp/error-messages/compiler-errors-2/compiler-error-c3001

Let me know if you need more infos :)

Thank you.

**Comment by kornelski** [comment]
Oh, that's disappointing. It looks like Visual Studio doesn't have full support for OpenMP.  I'll have a look at it later.

Since the --fast crash is gone, I'm going to close this issue. Thanks for your help.

**Comment by jalba69** [comment]
Hi, no worries then, it works and that's good enough for now. While it will be nice to have a faster process i can live without it.

Tested a bit more yesterday at night and no problems, had in mind to test a bit more today and report back but did not had any time for this yet, came home not long ago.

I see OpenMp related options in VS2017 not sure how they are supported, this is rather new to me and i'm not a coder/compiler guy.
I am downloading _Intel® IPP Library_ to integrate it inside VS2017, not sure if it will make any difference, probably not without changing some code parts but i'll try...

In the next days i will try to ask on the Rust-FFmpeg page and see if there is a way to have this thing built on windows so i could give another try as Gifski with video support which is what i'll miss the most.

Thanks again for all the help and sorry to have bothered you with all this.
